dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/scalar/JsonUpdateTest.java

```diff
@@ -0,0 +1,190 @@
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":null}";
+        String jPath = "$.k2";
+        String updateValue = null;
+        Assert.assertEquals(expectedJsonEvent, jsonUpdate.eval(jsonEvent, jPath, updateValue));
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNewNestedKeyValue() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"key1\":\"value1\",\"key2\":\"value2\"}}";
+        String jPath = "$.k3";
+        Map<String, String> updateValue = new HashMap<String, String>() {{
+            put("key1", "value1");
+            put("key2", "value2");
+        }};
+        String actual = jsonUpdate.eval(jsonEvent, jPath, updateValue);
+        Assert.assertEquals(expectedJsonEvent, actual);
+    }
+
+    @Test
+    public void shouldReturnUpdatedJsonStringForNewNestedKeyValueInNestedPath() {
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"nk1\":\"nv1\",\"nk2\":\"nv2\"}}";
+        String expectedJsonEvent = "{\"k1\":null,\"k2\":\"v2\",\"k3\":{\"nk1\":{\"key1\":\"value1\",\"key2\":\"value2\"},\"nk2\":\"nv2\"}}";
+        String jPath = "$.k3.nk1";
+        Map<String, String> updateValue = new HashMap<String, String>() {{
+            put("key1", "value1");
+            put("key2", "value2");
+        }};
+        String actual = jsonUpdate.eval(jsonEvent, jPath, updateValue);
+        Assert.assertEquals(expectedJsonEvent, actual);
+    }
+
+    @Test
+    public void shouldThrowErrorWhenNullJsonEvent() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("json string can not be null or empty");
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = null;
+        String jPath = "$.k2";
+        String updateValue = "updatedValue";
+        jsonUpdate.eval(jsonEvent, jPath, updateValue);
+    }
+
+    @Test
+    public void shouldThrowErrorWhenNullJPath() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("json can not be null or empty");
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String jPath = null;
+        String updateValue = "updatedValue";
+        jsonUpdate.eval(jsonEvent, jPath, updateValue);
+    }
+
+    @Test
+    public void shouldThrowErrorWhenEmptyJPath() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("json can not be null or empty");
+        JsonUpdate jsonUpdate = new JsonUpdate();
+        String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";
+        String jPath = "";
+        String updateValue = "updatedValue";
+        jsonUpdate.eval(jsonEvent, jPath, updateValue);
+    }
+
```

Comment:
Is it possible to remove a key from the jsonEvent using this UDF ?

For example, given `String jsonEvent = "{\"k1\":null,\"k2\":\"v2\"}";`

is it possible to remove the key `k1` itself from the json, so that it becomes 
`"{\"k2\":\"v2\"}"`

Reply:
Yes, it is possible we can introduce one overloaded method in the same UDF with `eval(String jsonEvent, String jPath)`  however it will not justify the action to the name of the UDF, like `JsonUpdate(jsonEvent, '$.k1')` and it rather adds confusion.  Instead I have added one more UDF JsonDelete can be used like `JsonDelete(jsonEvent, '$.k1')` which provides proper action to the function name.

Reply:
ok